clang-tools-extra/clangd/unittests/CodeCompleteTests.cpp

```diff
@@ -4449,6 +4458,14 @@ TEST(CompletionTest, SkipExplicitObjectParameter) {
                 ElementsAre(AllOf(named("bar"), signature("(int arg)"),
                                   snippetSuffix(""))));
   }
+  {
+    auto Result = codeComplete(testPath(TU.Filename), Code.point("c3"),
+                               Preamble.get(), Inputs, Opts);
+    EXPECT_THAT(
+        Result.Completions,
+        ElementsAre(AllOf(named("foo"), signature("<class self:auto>(int arg)"),
+                          snippetSuffix("<${1:class self:auto}>"))));
```

Comment:
Multiple things about this seem to be wrong:

 1. We should not get an explicit template parameter list in the snippet, because the template parameter can be deduced from the first argument. (We're expecting the user to write `(&A::foo)(a, 42);` most of the time, not `(&A::foo<A>)(a, 42);`).
 2. The snippet suffix is missing function parameters altogether.
 3. The signature should have two function parameters, not one.

It's fine to assert the current (buggy) behaviour, but please add a comment describing why it's wrong, and what we should get instead.